Detectors/ITSMFT/ITS/tracking/include/ITStracking/Road.h

```diff
@@ -41,30 +42,68 @@ class Road final
   void setFakeRoad(const bool);
   GPUhdni() int& operator[](const int&);
 
-  void resetRoad();
-  void addCell(int, int);
+  GPUhd() void resetRoad()
+  {
+    for (int i = 0; i < maxRoadSize; i++) {
+      mCellIds[i] = constants::its::UnusedIndex;
+    }
+    mRoadSize = 0;
+  }
 
-  static constexpr int mMaxRoadSize = 13;
+  GPUhd() void addCell(int cellLayer, int cellId)
+  {
+    if (mCellIds[cellLayer] == constants::its::UnusedIndex) {
+      ++mRoadSize;
+    }
+
+    mCellIds[cellLayer] = cellId;
+  }
+
+  // static constexpr int maxRoadSize = 13;
 
  private:
-  int mCellIds[mMaxRoadSize];
+  int mCellIds[maxRoadSize];
   int mRoadSize;
   int mLabel;
   bool mIsFakeRoad;
```

Comment:
while you are at it why not changing mRoadSize to a char, swapping the order with mLabel, and so that this bool does not take 32 bits due to padding? I.e.:

```c++
int mLabel;
char mRoadSize;
bool isFakeRoad;
```

which is 8 bytes or even better if the number of labels is small:

```c++
short mLabel; // max 2^16 labels
char mRoadSize;
bool mIsFakeRoad;
```

which is just 4 bytes.

Reply:
> while you are at it why not changing mRoadSize to a char, swapping the order with mLabel, and so that this bool does not take 32 bits due to padding? I.e.:
> 
> ```c++
> int mLabel;
> char mRoadSize;
> bool isFakeRoad;
> ```
> 
> which is 8 bytes or even better if the number of labels is small:
> 
> ```c++
> short mLabel; // max 2^16 labels
> char mRoadSize;
> bool mIsFakeRoad;
> ```
> 
> which is just 4 bytes.

Indeed `mRoadSize` can be changed to char or unsigned char and swapped, to further reduce the size of `Road<5>`, good catch.
This would shrink the total size from 32 to 28. I am also checking whether we still use the label. This would reduce the total size down to 24, leaving just 2 bytes to padding.
Updates after local test.

Reply:
Tested with local simulation, no problems detected, worth a round of CI.